### ISSUE TITLE
[Bug]: Lora's effect is different from A1111 In the refiner

### ISSUE THREAD

**Issue by sicatt** [lllyasviel/stable-diffusion-webui-forge#161]
### Checklist

- [X] The issue exists after disabling all extensions
- [X] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [X] The issue exists in the current version of the webui
- [ ] The issue has not been reported before recently
- [X] The issue has been reported before but has not been fixed yet

### What happened?

![at](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/67177866/08e7d07b-15a7-4cea-b229-a41426fe687c)

The effect of lora on refiner is different between forge and A1111.

### Steps to reproduce the problem

Here is the Lora we use
https://civitai.com/models/299418/dark-magician-girl-yu-gi-oh?modelVersionId=336299

Here are the checkpoints we use  animagine-xl-v3
https://civitai.com/models/260267/animagine-xl-v3

Here are the checkpoints we use  Pony Diffusion V6 XL 
https://civitai.com/models/257749?modelVersionId=290640

Prompts used
---------------------------
<lora:blamagiXLv1:1>, blonde hair, green eyes, long hair, medium breasts,blush stickers,wizard hat,

1girl,solo,

standing,
indoors,

(best quality:1), 
-------------------------

Negative prompts used
-------------------------
lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, artist name,  
-------------------------

Seed is 4078101471

Sampling method is DPM++2M Karras

image Size Width 920  Height 1152

CFG Scale 7

### What should have happened?

The effect of lora is also applied to the model specified in the refiner. That would generate the same picture as a1111.

### What browsers do you use to access the UI ?

Google Chrome

### Sysinfo

[sysinfo-2024-02-10-05-09.json](https://github.com/lllyasviel/stable-diffusion-webui-forge/files/14227992/sysinfo-2024-02-10-05-09.json)

### Console logs

```Shell
Loading weights [67ab2fd8ec] from G:\stable-diffusion-webui-forge\models\Stable-diffusion\ponyDiffusionV6XL_v6StartWithThisOne.safetensors
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
loaded straight to GPU
To load target model SDXL
Begin to load 1 model
Loading VAE weights specified in settings: G:\stable-diffusion-webui-forge\models\VAE\sdxl.vae.safetensors
To load target model SDXLClipModel
Begin to load 1 model
Moving model(s) has taken 0.63 seconds
Model loaded in 12.3s (unload existing model: 3.9s, load weights from disk: 0.1s, forge load real models: 7.2s, load VAE: 0.3s, calculate empty prompt: 0.7s).
To load target model SDXLClipModel
Begin to load 1 model
unload clone 0
Moving model(s) has taken 0.86 seconds
To load target model SDXL
Begin to load 1 model
unload clone 1
Moving model(s) has taken 4.01 seconds
 20%|████████▌                                  | 6/30 [00:00<00:03,  7.14it/s]Loading weights [1449e5b0b9] from G:\stable-diffusion-webui-forge\models\Stable-diffusion\animagine-xl-3.0.safetensors
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
Loading VAE weights specified in settings: G:\stable-diffusion-webui-forge\models\VAE\sdxl.vae.safetensors
To load target model SDXLClipModel
Begin to load 1 model
Moving model(s) has taken 0.74 seconds
Model loaded in 6.6s (unload existing model: 0.9s, forge load real models: 4.4s, load VAE: 0.3s, calculate empty prompt: 0.9s).
To load target model SDXL
Begin to load 1 model
Moving model(s) has taken 1.34 seconds
100%|██████████████████████████████████████████| 30/30 [00:12<00:00,  2.44it/s]
To load target model AutoencoderKL█████████████| 30/30 [00:12<00:00,  7.22it/s]
Begin to load 1 model
Moving model(s) has taken 0.60 seconds
Total progress: 100%|██████████████████████████| 30/30 [00:13<00:00,  2.17it/s]
Total progress: 100%|██████████████████████████| 30/30 [00:13<00:00,  7.22it/s]
```

### Additional information

I know that a few hours ago a correction was made regarding lora. But even after that fix came in, this problem was not resolved.

**Comment by lllyasviel** [comment]
update and test again

**Comment by sicatt** [comment]
The resolution of the problem has been confirmed! Thank you very much!

**Comment by Vigilence** [comment]
@sicatt can you confirm your lora is working with refiners with latest forge update?  I get lora errors when used with the xl refiner model on latest release but not if I use a commit from over a week ago.

**Comment by sicatt** [comment]
@Vigilence
I have been able to use Lora on refiner since this fix with no problems, including up to the latest update. The usage is Pony-v6 to Animagine-v3.

**Comment by Vigilence** [comment]
@sicatt  Can you test a lora with the original xl refiner model and let me know if it works?

**Comment by sicatt** [comment]
@Vigilence 
I set up the original xl refiner model as a refiner and tried the Lora effect. Indeed, there seems to be no Lora effect, but I felt that this could be caused by Lora, not forge. This is because Lora has a very different effect depending on the combination of which model it was trained on and which model it is used on.
My guess is that Lora trained on animaginelx-v3 would have no effect at all on the original xl refiner model.

The reason is that I tested the effect of Lora with the original xl refiner model set to the main model instead of the refiner, and the effect of Lora was not demonstrated. This means that the FORGE refiner is not the cause.

To prove this, I would try to create Lora with the original xl refiner model as the base model. If Lora is created with the original xl refiner model as the base model, the effect of Lora will be demonstrated even if the original xl refiner model is set as the refiner.

**Comment by Vigilence** [comment]
@sicatt Did you see an error in the cmd window?  If you update to the latest forge you should see an error message about the `Lora key not loading`

**Comment by sicatt** [comment]
@Vigilence 
Oops, I did indeed just try the latest version and it seems that Lora is not taking effect on the refiner.
I'm also seeing logs that look like errors that weren't showing up when things were going well.

**Comment by Vigilence** [comment]
@sicatt Thank you for checking!  I think the errors could be related to only the xl refiner model and Lora’s as they are not meant to work together (lora is not trained on the base xl refiner model). 

I think the errors should not appear if you use a different refiner like Pony-v6 or Animagine-v3 with the same Lora’s but I am not sure.

**Comment by sicatt** [comment]
@Vigilence 
No, I can confirm that even Lora, the blue-clad, blonde-haired magical girl I am presenting in this issue, is not effective. So the base model is ponev6 and the refiner is animagineXL-v3.
It seems that the problem, which was solved once in this issue, is occurring again.

Is there currently a new issue open regarding this issue? Is there any plan to fix it?

**Comment by Vigilence** [comment]
@sicatt @lllyasviel I have one [here](https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/194) but it’s been ignored.

**Comment by sicatt** [comment]
@Vigilence OK. I understand the situation.

**Comment by lllyasviel** [comment]
update
i just checked
there is no problem here. all results same with original webui.
please add different results from webui and forge, and steps to reproduce if user believe there is any problem

**Comment by sicatt** [comment]
@lllyasviel @Vigilence 
I have updated both A1111 and FORGE to the latest versions and tested them again, and was still able to reproduce the bug.
llyasviel, I'm sorry, but could you please check again using the following procedure?


**■Model and Lora to be used**
Here is the Lora we use
https://civitai.com/models/299418/dark-magician-girl-yu-gi-oh?modelVersionId=336299

Here are the checkpoints we use animagine-xl-v3
https://civitai.com/models/260267/animagine-xl-v3

Here are the checkpoints we use Pony Diffusion V6 XL
https://civitai.com/models/257749?modelVersionId=290640


**■Model Setup**
main model = **Pony Diffusion V6 XL**
refiner = **animagine-xl-v3**
VAE = None
refiner switch at = 0.2

**■Prompts used**
 <<lora:blamagiXLv1:1>>, blonde hair, green eyes, long hair, medium breasts,blush stickers,wizard hat,

1girl,solo,

standing,
indoors,

(best quality:1),

**■Negative prompts used**
lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, artist name,


**■Other Settings**
Seed is 4078101471

Sampling method is DPM++2M SDE Karras

image Size Width 920 Height 1152

CFG Scale 7


**■log**
Loading weights [67ab2fd8ec] from Q:\stable-diffusion-webui-forge\models\Stable-diffusion\ponyDiffusionV6XL_v6StartWithThisOne.safetensors[00:05<00:00,  6.78it/s]
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
loaded straight to GPU
To load target model SDXL
Begin to load 1 model
To load target model SDXLClipModel
Begin to load 1 model
Moving model(s) has taken 0.92 seconds
Model loaded in 9.1s (unload existing model: 2.0s, load weights from disk: 0.1s, forge load real models: 5.9s, calculate empty prompt: 1.0s).
To load target model SDXLClipModel
Begin to load 1 model
unload clone 0
Moving model(s) has taken 2.06 seconds
To load target model SDXL
Begin to load 1 model
unload clone 1
Moving model(s) has taken 3.85 seconds
 20%|█████████████████████████▏                                                                                                    | 6/30 [00:01<00:03,  6.22it/s]Loading weights [1449e5b0b9] from Q:\stable-diffusion-webui-forge\models\Stable-diffusion\animagine-xl-3.0.safetensors              | 6/30 [00:00<00:03,  7.11it/s]
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
loaded straight to GPU
To load target model SDXL
Begin to load 1 model
To load target model SDXLClipModel
Begin to load 1 model
Moving model(s) has taken 0.74 seconds
Model loaded in 7.4s (unload existing model: 1.0s, forge load real models: 5.4s, calculate empty prompt: 0.8s).
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:12<00:00,  2.43it/s]
To load target model AutoencoderKL████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:12<00:00,  6.74it/s]
Begin to load 1 model
Total progress: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:12<00:00,  2.34it/s]
Total progress: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:12<00:00,  6.74it/s]


![at](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/67177866/6b38ff05-4e41-4c94-bc87-ebe64a230246)

**Comment by doomengineer** [comment]
This is still happening for me.
